### ISSUE TITLE
select count(*); returns 1

### ISSUE THREAD

**Issue by swingbit** [MonetDB/MonetDB#7369]
**Describe the bug**
I just noticed that `SELECT count(*);` returns 1. It doesn't really make sense though. All DBMSs I tested give back a syntax error.

It's a pretty minor bug, I would say, though potentially misleading: I found out by forgetting a `FROM` in `SELECT count(*) FROM t;`:
```
sql>select count(*) t;
+------+
| t    |
+======+
|    1 |
+------+
1 tuple
```

What happened is that `t` has become an alias to the result of `select count(*)`. Of course this was my fault, but it induced me to think that the count of `t` was 1 and only after a careful look I noticed the mistake.

Bottom line though is that `SELECT count(*);` shouldn't be allowed - I think - as a valid statement.

**To Reproduce**
```
sql>select count(*);
+------+
| %2   |
+======+
|    1 |
+------+
1 tuple
```

**Expected behavior**
Syntax error

**Software versions**
 - MonetDB version number 11.43.26
 - OS and version: Fedora
 - self-installed and compiled

**Comment by joerivanruth** [comment]
I'm not sure about that. Conceptually, if you don't specify a FROM clause you get an implicit 1-row table to SELECT from. So `SELECT 42` computes the expression 42 for each row of that implicit one-row table. 
Older versions of Oracle made the FROM clause mandatory and provided a special one-row one-column table `dual` for convenience so you could write `SELECT 42 FROM DUAL`. There was much rejoicing when this limitation was finally removed. See [DUAL table](https://en.wikipedia.org/wiki/DUAL_table) on Wikipedia.

The semantics of FROM-less SELECTs seem fairly straightforward and I would not be in favor of special-casing this and making the SQL even more irregular.

**Comment by swingbit** [comment]
I see. I actually didn't know about `DUAL` and the implicit 1-row table (I've read up about it now). I had always interpreted `SELECT 42` as projecting a constant (with no table), rather than projecting 42 over an implicit 1-row table. It does make sense.

It's up to you. My only remark is that - reasoning from a user's perspective -  while `SELECT 42` is perfectly intuitive and is used in practice, `SELECT count(*)` is definitely not intuitive and I can't see any reasonable use for it.

**Comment by mvdvm** [comment]
Standard SQL requires a `FROM table` part in a SELECT query. To allow to query session variables such as CURRENT_USER, CURRENT_TIMESTAMP, CURRENT_SCHEMA some RDBMS have added a virtual system table of 1 row and 1 column for convenience. Oracle was the first and called it DUAL. Other RDBMS followed, see: [In other database systems](https://en.wikipedia.org/wiki/DUAL_table#In_other_database_systems) but with different table names.

The option to have **no** `FROM table` part in a SELECT query is a MonetDB extension of the SQL standard. This is also supported by some other RDBMS as it is simpler to use (less typing) and more portable than `FROM DUAL` or `FROM sysibm.sysdummy1`.
However in order to get a result from a tableless query, it must behave as if it was querying a single row table.
Hence `SELECT count(*)` should behave the same as `SELECT count(*) from DUAL` in Oracle which returns 1.
Note that MonetDB does **not** support  `SELECT *`, where `SELECT * from DUAL` in Oracle is allowed.

It is debatable if it should return an error instead.
Changing `SELECT count(*)` result may have negative impact for existing scripts and users.
